src/System.Management.Automation/engine/MshObject.cs

```diff
@@ -2302,17 +2317,173 @@ DynamicMetaObject IDynamicMetaObjectProvider.GetMetaObject(Expression parameter)
+        internal bool HasGeneratedReservedMembers
+        {
+            get => _flags.HasFlag(PSObjectFlags.HasGeneratedReservedMembers);
+            set
+            {
+                if (value)
+                {
+                    _flags |= PSObjectFlags.HasGeneratedReservedMembers;
+                }
+                else
+                {
+                    _flags &= ~PSObjectFlags.HasGeneratedReservedMembers;
+                }
+            }
+        }
 
-        #endregion
+        internal bool ImmediateBaseObjectIsEmpty
+        {
+            get => _flags.HasFlag(PSObjectFlags.ImmediateBaseObjectIsEmpty);
+            set
+            {
+                if (value)
+                {
+                    _flags |= PSObjectFlags.ImmediateBaseObjectIsEmpty;
+                }
+                else
+                {
+                    _flags &= ~PSObjectFlags.ImmediateBaseObjectIsEmpty;
+                }
+            }
+        }
+
+
+        internal WriteStreamType WriteStream
+        {
+            get => _writeStream;
+            set => _writeStream = value;
+        }
+
+        /// <summary>
+        /// Members from the adapter of the object before it was serialized
+        /// Null for live objects but not null for deserialized objects.
+        /// </summary>
+        internal PSMemberInfoInternalCollection<PSPropertyInfo> AdaptedMembers
+        {
+            get => _adaptedMembers;
+            set => _adaptedMembers = value;
+        }
+
+        internal static DotNetAdapter DotNetStaticAdapter => s_dotNetStaticAdapter;
+
+        internal static PSTraceSource MemberResolution => s_memberResolution;
+
+        /// <summary>
+        /// Members from the adapter of the object before it was serialized
+        /// Null for live objects but not null for deserialized objects.
+        /// </summary>
+        internal PSMemberInfoInternalCollection<PSPropertyInfo> ClrMembers
+        {
+            get => _clrMembers;
+            set => _clrMembers = value;
+        }
+
+        internal static DotNetAdapter DotNetInstanceAdapter => s_dotNetInstanceAdapter;
+
+        /// <summary>
+        /// Gets an instance member if it's name matches the predicate. Otherwise null.
+        /// </summary>
+        internal PSPropertyInfo GetFirstPropertyOrDefault(MemberNamePredicate predicate)
+        {
+            if (_instanceMembers != null)
+            {
+                foreach (var instanceMember in _instanceMembers)
+                {
+                    if (instanceMember is PSPropertyInfo propInfo
+                        && predicate.Invoke(propInfo.Name))
+                    {
+                        return propInfo;
+                    }
+                }
+            }
+
+            return Properties.GetFirstOrDefault(predicate) as PSPropertyInfo;
+        }
+
+        [Flags]
+        enum PSObjectFlags : byte
+        {
+            None = 0,
+            /// <summary>
```

Comment:
Please add new line.
Below too.

Reply:
It is a private enum, so there is no "must" here. Not sure what value it would provide.

Reply:
Please use explicit `private enum PSObjectFlags : byte`.
My previous comment was about bad formatting the code. I expect that there is CodeFactor issues.